WFInfo/Login.xaml.cs

```diff
@@ -1,124 +1,159 @@
-        {
-            if (Email.Text == "Email")
-                Email.Text = "";
-        }
-
-        /// <summary>
-        /// Allow the window to be spawned in an appropriate place.
-        /// </summary>
-        /// <param name="x">Left most border of the window</param>
-        /// <param name="y">Top most border of the window</param>
-        public void MoveLogin(double x, double y)
-        {
-            Left = x;
-            Top = y;
-            Show();
-        }
-    }
-}
+﻿using System;
+using System.Runtime.InteropServices;
+using System.Windows;
+using System.Windows.Input;
+using Google.Apis.Auth;
+using Microsoft.Win32;
+using System.Windows.Media;
+
+namespace WFInfo
+{
+    /// <summary>
+    /// Interaction logic for Login.xaml
+    /// </summary>
+    public partial class Login : Window
+    {
+
+        #region default methods
+        public Login()
+        {
+            InitializeComponent();
+        }
+
+        /// <summary>
+        /// Hides the window 
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void HideExternal(object sender, MouseButtonEventArgs e)
+        {
+            Main.searchBox.Hide(); //hide search it if the user minimizes the window even though not yet logged in
+            Hide();
+        }
+
+        /// <summary>
+        /// Allows the window to be dragged
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private new void MouseDown(object sender, MouseButtonEventArgs e)
+        {
+            if (e.ChangedButton == MouseButton.Left)
+                DragMove();
+        }
+        #endregion
+
+        /// <summary>
+        /// Attempts to log in with the filled in credentials.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private async void LoginClick(object sender, MouseButtonEventArgs e)
+        {
+
+            if (!Main.dataBase.IsJwtAvailable())
+            {
+                try
+                {
+                    await Main.dataBase.GetUserLogin(Email.Text, Password.Password);
+                }
+                catch (Exception ex)
+                {
+                    Main.dataBase.JWT = null;
+                    Settings.Save();
+                    Main.AddLog("Couldn't login: " + ex);
+                    var StatusMessage = "Status"; //StatusMessage = text to display on StatusUpdate() AND the error box under login 
+                    var StatusSeverity = 0; //StatusSeverity = Severity for StatusUpdate()
+                    if (ex.Message.Contains("email"))
+                    {
+                        if (ex.Message.Contains("app.form.invalid"))
+                        {
+                            StatusMessage = "Invalid email form";
+                            StatusSeverity = 2;
+
+                        }
+                        else

```

Comment:
if else is not properly structured, results that the "else" state always gets executed and thus "invalid email form" never gets properly used.

Reply:
how is it wrong in that case ? cause double enter ?

Reply:
The "else" statement without any curly brackets only executes the next line of code. The following code is no longer seen as the "else" statement and will always be ran. It was the severity that didn't get changed not the message